pavelib/utils/decorators.py

```diff
@@ -0,0 +1,53 @@
+""" a collection of utililty decorators for paver tasks """
+
+import multiprocessing
+from functools import wraps
+
+
+def timeout(limit=60):
+    """
+    kill a function if it has not completed within a specified timeframe
+    """
+
+    def _handle_threaded_function(*args, **kwargs):
```

Comment:
You mention threads a few times, but then actually use a separate process instead in the code.

Reply:
Ah yes. I had originally used the threading library, but getting a functions return value proved difficult and switched to multiprocessing. Thanks for catching that.